tests/test_gail.py

```diff
@@ -38,15 +38,31 @@ def test_gail(expert_env):
             obs = env.reset()
     del dataset, model
 
-
-def test_generate_pendulum():
-    model = SAC('MlpPolicy', 'Pendulum-v0', verbose=0)
-    generate_expert_traj(model, 'expert_pendulum', n_timesteps=1000, n_episodes=10)
-
-
-def test_generate_cartpole():
-    model = DQN('MlpPolicy', 'CartPole-v1', verbose=0)
-    generate_expert_traj(model, 'expert_cartpole', n_timesteps=1000, n_episodes=10)
+@pytest.mark.parametrize("generate_env", [(SAC, 'MlpPolicy', 'Pendulum-v0', 1),
+                                            (DQN, 'MlpPolicy', 'CartPole-v1', 1),
+                                            (A2C, 'MlpPolicy', 'Pendulum-v0', 1),
+                                            (ACER, 'MlpPolicy', 'CartPole-v1', 1),
+                                            (PPO2, 'CnnPolicy', 'BreakoutNoFrameskip-v4', 1),
+                                            (A2C, 'MlpLstmPolicy', 'Pendulum-v0', 1),
+                                            (ACER, 'MlpLstmPolicy', 'CartPole-v1', 1),
+                                            (PPO2, 'CnnLstmPolicy', 'BreakoutNoFrameskip-v4', 1),
+                                            (A2C, 'CnnPolicy', 'BreakoutNoFrameskip-v4', 8),
+                                            (PPO2, 'CnnLstmPolicy', 'BreakoutNoFrameskip-v4', 8)])
+
+def test_generate(generate_env):
+    model, policy, env_name, n_env = generate_env
+
+    if n_env > 1:
+        env = make_atari_env(env_name, num_env=n_env, seed=0)
+        model = model(policy, env, verbose=0)
+    else:
+        try:
+            model = model(policy, env_name, verbose=0, nminibatches=1)
```

Comment:
Are you doing that for recurrent policies ?
Also to reduce time taken for LSTM policies, please use A2C for them (the default `n_steps` is quite small compared to PPO2 or ACER)
You don't need to test all algorithms, this is not the point of the test here, you should test:
- 1 discrete action (CartPolve-v1) env with both MLP and LSTM
- 1 continuous action  (Pendulum-v0) env with both MLP and LSTM

`CnnLstmPolicy` and `CnnPolicy` have the same API as `MlpPolicy` and `MlpLstmPolicy` so you can remove those tests (except if @hill-a or @erniejunior disagree)

Generating images (with the Breakout env) is quite time consuming and is already tested in the `test_pretrain_image` test. If you keep tests with `Breakout``then you must use `n_episodes=1` otherwise, it will take forever.

Reply:
ok, I have changed it to five shorter tests.

Reply:
Good, the CI test time is better now